pkg/resource/configurator.go

```diff
@@ -75,3 +75,28 @@ func kindish(obj runtime.Object) string {
 	}
 	return strings.ToLower(reflect.TypeOf(obj).Elem().Name())
 }
+
+type ObjectMetaConfiguratorWithNewNamingFormat struct {
+	typer runtime.ObjectTyper
+}
+
+// NewObjectMetaConfigurator returns a new ObjectMetaConfigurator.
+func NewObjectMetaConfiguratorWithNewNamingFormat(t runtime.ObjectTyper) *ObjectMetaConfiguratorWithNewNamingFormat {
+	return &ObjectMetaConfiguratorWithNewNamingFormat{typer: t}
+}
+
+// Configure the supplied Managed resource's object metadata.
+func (c *ObjectMetaConfiguratorWithNewNamingFormat) Configure(_ context.Context, cm Claim, cs NonPortableClass, mg Managed) error {
+	mg.SetNamespace(cs.GetNamespace())
+	if mg.GetName() == "" {
```

Comment:
Do we need this check? I recall that configurators are applied in order, so if a resource claim implementation wanted to use this configurator but also specify a bespoke name it could do so by applying this configurator first, then overriding its behaviour in a subsequent configurator.

Reply:
I was under impression that already existing resource might come through this configurator and we won't want to set `GenerateName` for an existing resource, but I see that's not the case. It's called only if the resource doesn't exist. Removed the check.